projects/core/src/lib/layers/legend/legend-raster/raster-legend.component.ts

```diff
@@ -56,6 +56,52 @@ export class CastMeasurementToContinuousPipe implements PipeTransform {
     }
 }
 
+export function unifyDecimals(values: number[]): number[] {
+    // Early return if all values differ by more than 1
+    if (values.every((x) => x >= 1)) {
+        const toCompare: number[] = values.map((x) => Math.floor(x)).sort((a, b) => a - b);
+        let oneApart = true;
+        for (let i = 0; i < toCompare.length - 1; i++) {
+            if (toCompare[i + 1] - toCompare[i] < 1) {
+                oneApart = false;
+            }
+        }
+        if (oneApart) return values.map((x) => Math.floor(x));
+    }
+
+    // Find highest overlap for cut-off point
+    let maxOverlap = 0;
+    values.forEach((x) => {
+        values.forEach((y) => {
+            if (x === y) {
+                return;
+            }
+            const overlap: number = overlappingDigits(x, y);
+            maxOverlap = overlap > maxOverlap ? overlap : maxOverlap;
+        });
+    });
+    return values.map((x) => {
+        const preDecimals = Math.floor(x).toString().length;
+        const roundAt = Math.pow(10, Math.max(0, maxOverlap + 2 - preDecimals));
+        return Math.round(x * roundAt) / roundAt;
+    });
+}
+
+export function overlappingDigits(val1: number, val2: number): number {
+    let overlap = 0;
+    const str1 = val1.toString();
+    const str2 = val2.toString();
+    const maxLength = Math.min(str1.length, str2.length);
+    let passedDecimal = false;
+    while (overlap < maxLength && str1.charAt(overlap) === str2.charAt(overlap)) {
+        if (str1.charAt(overlap) === '.' && str2.charAt(overlap) === '.') {
+            passedDecimal = true;
+        }
+        overlap++;
+    }
```

Comment:
Can't you make a for loop from 0 to maxLength and break early?

Reply:
Changed the while-loop to a for-loop with an early break if any difference is found. 330864ceb13c28994b236530b8cf6744a8428b7f